core/src/main/java/org/apache/accumulo/core/metadata/StoredTabletFile.java

```diff
@@ -151,7 +150,11 @@ public String toString() {
    * Validates that the provided metadata string for the StoredTabletFile is valid.
    */
   public static void validate(String metadataEntry) {
+    final TabletFileCq tabletFileCq = deserialize(metadataEntry);
+    // Validate the path
     ReferencedTabletFile.parsePath(deserialize(metadataEntry).path);
+    // Validate the range
+    requireRowRange(tabletFileCq.range);
```

Comment:
This check should always pass because the de-serialization is always creating a range object using rows  and not keys.  Wonder if instead of always checking this at runtime, the validation of this assumption could be moved to a unit test somehow.

Reply:
I went ahead and merged this but I like your idea of moving it to a unit test. If we can guarantee that the construction of the TabletFileFq will always be valid with testing then we could remove the check. I'll create a follow on issue to add some tests for this and then I can remove this validation.

Reply:
Follow on issue created for adding tests and removing the validation https://github.com/apache/accumulo/issues/3636